zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/NoteInterpreterLoader.java

```diff
@@ -192,4 +192,16 @@ public Interpreter get(String replName) {
 
     throw new InterpreterException(replName + " interpreter not found");
   }
+
+  private InterpreterSetting getDefaultInterpreterSetting(List<InterpreterSetting> settings) {
+    return settings.get(0);
+  }
+
+  Optional<InterpreterSetting> getDefaultInterpreterSetting() {
+    List<InterpreterSetting> settings = getInterpreterSettings();
+    if (settings == null || settings.isEmpty()) {
+      return Optional.absent();
+    }
+    return Optional.of(settings.get(0));
+  }
```

Comment:
Why do you make this method to return optional, not just returning getDefaultInterpreterSetting(getInterpreterSettings())? It looks like that Note.getDefaultInterpreterName uses this method only. If you want to check nullable list and emptiness, should you check the above method of this method?